rusk-wallet/src/wallet/transaction.rs

```diff
@@ -532,11 +533,23 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
 
         let stake_pk = BlsPublicKey::from(&stake_sk);
 
-        let reward_amount = state
+        let available_reward = state
             .fetch_stake(&stake_pk)
             .await?
             .map(|s| s.reward)
-            .unwrap_or_default();
+            .ok_or(Error::NoReward)?;
+
+        let reward_amt_withdrawn = if let Some(reward_amt) = reward_amt {
+            // throw error if we try to withdraw more than available
+            if reward_amt > available_reward {
+                return Err(Error::NotEnoughReward);
+            }
+
+            *reward_amt
+        } else {
+            // withdraw all the reward if no amt specified to withdraw
+            available_reward
+        };
```

Comment:
I think this gets more readable with a match statement:
```suggestion
        let reward_amt_withdrawn = match reward_amt {
            // if user specified an amount to withdraw we check if it's
            // available
            Some(reward_amt) => {
                if reward_amt > available_reward {
                    return Err(Error::NotEnoughReward);
                }

                *reward_amt
            }
            // withdraw all the reward if no amt specified to withdraw
            None => *available_reward,
        };
```